server/events/yaml/raw/project.go

```diff
@@ -106,10 +115,31 @@ func validProjectName(name string) bool {
 
 func validApplyReq(value interface{}) error {
 	reqs := value.([]string)
-	for _, r := range reqs {
-		if r != ApprovedApplyRequirement && r != MergeableApplyRequirement && r != UnDivergedApplyRequirement {
-			return fmt.Errorf("%q is not a valid apply_requirement, only %q, %q and %q are supported", r, ApprovedApplyRequirement, MergeableApplyRequirement, UnDivergedApplyRequirement)
+	for _, req := range reqs {
+		if _, ok := applyRequirements[req]; !ok {
+			return fmt.Errorf("%q is not a valid apply_requirement, only %s are supported", req, buildValidApplyRequirementsString())
 		}
 	}
 	return nil
 }
+
+func buildValidApplyRequirementsString() string {
+	// Sort keys.
+	applyRequirementsList := make([]string, 0, len(applyRequirements))
+	for applyReq := range applyRequirements {
+		applyRequirementsList = append(applyRequirementsList, applyReq)
+	}
+	sort.Strings(applyRequirementsList)
+
+	var returnString strings.Builder
+	counter := 0
+	for _, applyReq := range applyRequirementsList {
+		if counter == len(applyRequirements)-1 {
+			returnString.WriteString(fmt.Sprintf("and %q", applyReq))
+			break
+		}
+		returnString.WriteString(fmt.Sprintf("%q, ", applyReq))
+		counter++
+	}
+	return returnString.String()
```

Comment:
lets just replace this logic in favor of: `strings.Join(applyRequirementsList, ",")` Also we can make the result of this function a global var since this never changes after startup.

The errorf statement above can then be "%q is not a valid apply_requirement, supported apply requirements are: ..."

Reply:
Done!